[0.71][Android] Update pages on New Architecture for 71 (#3453)
facebook/react-native-website: docs/new-architecture-app-intro.md

There are a few prerequisites that should be addressed before the New Architecture is enabled in your application.

> ## Uprate to the latest React Native version

Reviewer: ```suggestion
## Update to the latest React Native version
```

React Native released the support for the New Architecture with the release `0.68.0`.

> This guide is written with the expectation that you’re using the **latest** React Native release. At the moment of writing, this is `0.71.0`.

Reviewer 2: ```suggestion
This guide is written with the expectation that you’re using the [**latest** React Native release](https://github.com/facebook/react-native/releases/latest). 
```
In this way, we don't have to update the doc anymore! 😉

You can find instructions on how to upgrade in the page [upgrading to new versions](/docs/upgrading.md).

## Use Hermes

Hermes is an open-source JavaScript engine optimized for React Native. Hermes is enabled by default, and you have to explicitly disable it if you want to use JSC.

We highly recommend using Hermes in your application. With Hermes enabled, you can use the JavaScript debugger in Flipper to directly debug your JavaScript code.

> Please [follow the instructions on the hermes page](hermes) to learn how to enable/disable Hermes.

Reviewer: ```suggestion
Please [follow the instructions on the Hermes page](hermes) to learn how to enable/disable Hermes.
```

  }
```

</TabItem>
</Tabs>

### Update your Activity Delegate

Finally, update your `MainActivity.java` file by providing a React Activity Delegate:

<Tabs groupId="android-language" defaultValue={constants.defaultAndroidLanguage} values={constants.androidLanguages}>
<TabItem value="java">

```diff title="MainActivity.java"
+import com.facebook.react.defaults.DefaultReactActivityDelegate;

public class MainActivity extends ReactActivity {

+ @Override
+ protected ReactActivityDelegate createReactActivityDelegate() {
+   return new DefaultReactActivityDelegate(
+       this,
+       getMainComponentName(),
+       true, // Here we enable Fabric
+       true, // Here we enable Concurrent React Features
+       );
+ }

}
```

</TabItem>
<TabItem value="kotlin">

```diff title="MainApplication.kt"
+import com.facebook.react.defaults.DefaultReactActivityDelegate;

public class MainActivity : ReactActivity {

+ @Override
+ override protected fun createReactActivityDelegate() =
+   DefaultReactActivityDelegate(
+       this,
+       mainComponentName,
+       true, // Here we enable Fabric
+       true, // Here we enable Concurrent React Features
+   )

}
```

</TabItem>
</Tabs>

> 

Reviewer 2: ❤️  I love that all this code is going away! ❤️
